src/Relations/MorphToMany.php

```diff
@@ -0,0 +1,461 @@
+            $this->parent->pull($this->table, $data);
+
+            // Prepare the query to select all related objects.
+            if (count($ids) > 0) {
+                $query->whereIn($this->relatedKey, $ids);
+            }
+
+            // Remove the relation from the related.
+            $query->pull($this->foreignPivotKey, $this->parent->{$this->parentKey});
+        } else {
+            // Remove the relation from the parent.
+            $this->parent->pull($this->relatedPivotKey, $ids);
+
+            // Prepare the query to select all related objects.
+            if (count($ids) > 0) {
+                $query->whereIn($this->relatedKey, $ids);
+            }
+
+            // Remove the relation to the related.
+            $query->pull($this->table, [
+                [
+                    $this->foreignPivotKey => $this->parent->{$this->parentKey},
+                    $this->morphType => $this->parent->getMorphClass(),
+                ],
+            ]);
+        }
+
+        if ($touch) {
+            $this->touchIfTouching();
+        }
+
+        return count($ids);
+    }
+
+    /** @inheritdoc */
+    protected function buildDictionary(Collection $results)
+    {
+        $foreign = $this->foreignPivotKey;
+
+        // First we will build a dictionary of child models keyed by the foreign key
+        // of the relation so that we will easily and quickly match them to their
+        // parents without having a possibly slow inner loops for every models.
+        $dictionary = [];
+
+        foreach ($results as $result) {
+            if ($this->getInverse()) {
+                foreach ($result->$foreign as $item) {
+                    $dictionary[$item][] = $result;
+                }
+            } else {
+                // Collect $foreign value from pivot column of result model
+                $items = $this->extractIds($result->{$this->table} ?? [], $foreign);
+                foreach ($items as $item) {
+                    $dictionary[$item][] = $result;
+                }
+            }
+        }
+
+        return $dictionary;
+    }
+
+    /** @inheritdoc */
+    public function newPivotQuery()
+    {
+        return $this->newRelatedQuery();
+    }
+
+    /**
+     * Create a new query builder for the related model.
+     *
+     * @return \Illuminate\Database\Query\Builder
+     */
+    public function newRelatedQuery()
+    {
+        return $this->related->newQuery();
+    }
+
+    /**
+     * Get the fully qualified foreign key for the relation.
+     *
+     * @return string
+     */
+    public function getForeignKey()
+    {
+        return $this->foreignPivotKey;
+    }
+
+    /** @inheritdoc */
+    public function getQualifiedForeignPivotKeyName()
+    {
+        return str_contains($this->foreignPivotKey, '.')
+            ? $this->foreignPivotKey
+            : $this->table . '.' . $this->foreignPivotKey;
+    }
+
+    /** @inheritdoc */
+    public function getQualifiedRelatedPivotKeyName()
```

Comment:
Since the pivot column stores on `MorphedByMany` model, we just need to prefix the `$this->foreignPivotKey` with the pivot column name. On the other side, on `MorphToMany` model, the relationship's data will be stored as an array and doesn't have any nested column to be prefixed with.